selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/languageserver/TypeScriptEditingTest.java

```diff
@@ -0,0 +1,128 @@
+  private static final String PROJECT_NAME =
+      NameGenerator.generate(TypeScriptEditingTest.class.getSimpleName(), 4);
+  private static final String PATH_TO_GREETER_FILE = PROJECT_NAME + "/Greeter.ts";
+  private static final String PATH_TO_PRINT_TEST_FILE = PROJECT_NAME + "/printTest.ts";
+
+  @InjectTestWorkspace(template = ECLIPSE_NODEJS)
+  private TestWorkspace workspace;
+
+  @Inject private Ide ide;
+  @Inject private Menu menu;
+  @Inject private Wizard wizard;
+  @Inject private Consoles consoles;
+  @Inject private CodenvyEditor editor;
+  @Inject private CommandsPalette commandsPalette;
+  @Inject private ProjectExplorer projectExplorer;
+  @Inject private AskForValueDialog askForValueDialog;
+  @Inject private TestProjectServiceClient testProjectServiceClient;
+
+  @BeforeClass
+  public void setUp() throws Exception {
+    URL resource = getClass().getResource("/projects/type-script-simple-project");
+    testProjectServiceClient.importProject(
+        workspace.getId(), Paths.get(resource.toURI()), PROJECT_NAME, NODE_JS);
+    ide.open(workspace);
+
+    projectExplorer.waitVisibleItem(PROJECT_NAME);
+    projectExplorer.quickExpandWithJavaScript();
+    projectExplorer.openItemByPath(PATH_TO_GREETER_FILE);
+  }
+
+  @Test
+  public void checkMainFeaturesTypeScriptLS() {
+    String intitTypeScriptLanguageServerMessage =
+        String.format(
+            "Finished language servers initialization, file path '/%s'", PATH_TO_GREETER_FILE);
+
+    consoles.waitExpectedTextIntoConsole(intitTypeScriptLanguageServerMessage);
+    checkCodeValidation();
+    checkCodeAssistant();
+    checkGoToDefinition();
+  }
+
+  private void checkCodeValidation() {
+    final int expectedValueOfErrorMarkers = 9;
+    editor.waitActive();
+    editor.goToPosition(13, 2);
+    editor.typeTextIntoEditor(SPACE.toString());
+    final int actualValueErrorMarkers = editor.getMarkersQuantity(ERROR);
+    assertEquals(
+        actualValueErrorMarkers,
+        expectedValueOfErrorMarkers,
+        String.format(
+            "The expected value of errors marker should be %d but actual %d",
+            expectedValueOfErrorMarkers, actualValueErrorMarkers));
+    editor.waitMarkerInPositionAndMoveCursor(ERROR_OVERVIEW, 13);
+    editor.waitTextInToolTipPopup("Cannot find name 'c'");
+    editor.goToPosition(13, 2);
+    editor.typeTextIntoEditor(DELETE.toString());
+    editor.waitAllMarkersInvisibility(ERROR);
+  }
+
+  private void checkCodeAssistant() {
+    String textFromWholeCodeAssistantScope =
+        "AbortController\nAbortSignal\nabstract\nActiveXObject\naddEventListener\nalert";
+    String textFromGreeterObject = "greet\ngreeting\ntestPrint";
+
+    editor.goToPosition(28, 36);
+    editor.launchAutocomplete();
+    editor.waitTextIntoAutocompleteContainer(textFromWholeCodeAssistantScope);
+    editor.closeAutocomplete();
+    editor.typeTextIntoEditor(ENTER.toString());
+    editor.typeTextIntoEditor("greeter.");
```

Comment:
it's hard to understand test case here
Could you, please, name string "greeter." somehow to make its propose an obvious?

Reply:
In the test based on simple Greeter example like:
```typescript
class Greeter {
    greeting: string;
    constructor(message: string) {
        this.greeting = message;
    }
    greet() {
        return "Hello, " + this.greeting;
    }
    
     testPrint(): void {
        const printVar = new Print();
        printVar.print("test print");
    }
}

let greeter = new Greeter("world");
```
and i cannot see difficult parts for understanding

Reply:
Ok, your proposal?